smart_contracts/contracts_as/example/increment-counter/README.md

```diff
@@ -0,0 +1,136 @@
+
+
+### Step 1. Configuring the Index.ts File
+
+The first section of *assembly/index.ts* outlines the necessary dependencies to include and the constants to establish.
+
+1. Begin with `//@ts-nocheck` to disable semantic checks.
+
+2. Establish imported dependencies from other packages.
+
+3. Establish constants for use within the installed contract.
+
+```typescript
+
+// Importing necessary aspects of external packages.
+import * as CL from "../../../../contract_as/assembly";
+import { Error, ErrorCode } from "../../../../contract_as/assembly/error";
+import { fromBytesI32, fromBytesString, toBytesMap } from "../../../../contract_as/assembly/bytesrepr";
+import { Key } from "../../../../contract_as/assembly/key";
+import { CLValue, CLType, CLTypeTag } from "../../../../contract_as/assembly/clvalue";
+import { Pair } from "../../../../contract_as/assembly/pair";
+
+// Establishing constants for use within the installing contract.
+const COUNT_KEY = "count";
+const COUNTER_INC = "counter_inc";
+const COUNTER_GET = "counter_get";
+const COUNTER_KEY = "counter";
+const CONTRACT_VERSION_KEY = "version";
+
+```
+
+### Step 2. Defining the Contract Entry Points
+
+Each entry point that we later establish will require an associated function to perform the desired action. In the next section of *index.ts* we define these functions to perform the actions of incrementing the counter and viewing the current counter value.
+
+```typescript
+
+// Creating a function that will be used by the incrementing entry point.
+export function counter_inc(): void {
+  let countKey = CL.getKey(COUNT_KEY);
+  if (!countKey) {
+    Error.fromErrorCode(ErrorCode.MissingKey).revert();
+    return;
+  }
+
+  if (!countKey.isURef()) {
+    Error.fromErrorCode(ErrorCode.UnexpectedKeyVariant).revert();
+    return;
+  }
+
+  const one = CLValue.fromI32(1);
+
+  countKey.add(one);
+}
+
+// Creating a function that will be used by the value retrieval entry point.
+export function counter_get(): void {
+  let countKey = CL.getKey(COUNT_KEY);
+
+  if (!countKey) {
+    Error.fromErrorCode(ErrorCode.MissingKey).revert();
+    return;
+  }
+
+  if (!countKey.isURef()) {
+    Error.fromErrorCode(ErrorCode.UnexpectedKeyVariant).revert();
+    return;
+  }
+  let countData = countKey.read();
+  if (!countData) {
+    Error.fromErrorCode(ErrorCode.ValueNotFound).revert();
+    return;
+  }
+  let value = fromBytesI32(<Uint8Array>countData).unwrap();
+  CL.ret(CLValue.fromI32(value));
+}
+
+```
+
+### Step 3. Defining the `Call` Function
+
+We will also define the `Call` function that will start the code execution and perform the actual code installation. This function initializes the contract by setting the counter to 0 and creates `NamedKeys` in the current context. Specifically, it creates the keys `counter_package_name` and `counter_access_uref`.
```

Comment:
Also some changes around the wording of creating NamedKeys, I'd rather we say we create the NamedKeys to pass them along to the contract where they are persisted within the contracts context.